src/styles/_header.scss

```
@@ -37,5 +37,10 @@
     text-transform: uppercase;
     font-size: 12px;
     font-weight: 500;
+    transition: transform 0.3s;
+
+    &:hover {
+      color: #00acdc;
+    }
```

Здесь ты анимируешь color, а не transform
```suggestion
    transition: color 0.3s;

    &:hover {
      color: #00acdc;
    }
```